And when using raw string, but without the backticks, sphinx does still interprete the backslash? 
If that is the case, I think this is the best solution possible (it looks good in html, and better in the docstring compared to the double backslashes

The travis failure is because we have a linter for one of the 2.7 builds. If you look at the log of the failing build, down below, there is the output of the linter:

pandas/io/parsers.py:282:80: E501 line too long (80 > 79 characters)

--- a/pandas/core/frame.py
+++ b/pandas/core/frame.py
@@ -1306,7 +1306,7 @@ class DataFrame(NDFrame):
                quotechar='"', line_terminator='\n', chunksize=None,
                tupleize_cols=False, date_format=None, doublequote=True,
                escapechar=None, decimal='.', **kwds):
-        """Write DataFrame to a comma-separated values (csv) file
+        r"""Write DataFrame to a comma-separated values (csv) file
 
         Parameters
         ----------
@@ -1343,7 +1343,7 @@ class DataFrame(NDFrame):
             a string representing the compression to use in the output file,
             allowed values are 'gzip', 'bz2', 'xz',
             only used when the first argument is a filename
-        line_terminator : string, default '\\n'
+        line_terminator : string, default ``'\n'``
             The newline character or character sequence to use in the output
             file
         quoting : optional constant from csv module

--- a/pandas/io/parsers.py
+++ b/pandas/io/parsers.py
@@ -277,11 +277,11 @@ _engine_doc = """engine : {'c', 'python'}, optional
     Parser engine to use. The C engine is faster while the python engine is
     currently more feature-complete."""
 
-_sep_doc = """sep : str, default {default}
+_sep_doc = r"""sep : str, default {default}
     Delimiter to use. If sep is None, will try to automatically determine
-    this. Separators longer than 1 character and different from '\s+' will be
-    interpreted as regular expressions, will force use of the python parsing
-    engine and will ignore quotes in the data. Regex example: '\\r\\t'"""
+    this. Separators longer than 1 character and different from ``'\s+'`` will
+    be interpreted as regular expressions, will force use of the python parsing
+    engine and will ignore quotes in the data. Regex example: ``'\r\t'``"""
 
 _read_csv_doc = """
 Read CSV (comma-separated) file into DataFrame

